service/src/main/java/bio/terra/workspace/service/folder/FolderService.java

```diff
@@ -125,6 +139,8 @@ private void getResourcesInFolder(
                 if (StewardshipType.REFERENCED == resource.getStewardshipType()) {
                   referencedResources.add(resource);
                 } else if (StewardshipType.CONTROLLED == resource.getStewardshipType()) {
+                  controlledResourceMetadataManager.validateControlledResourceAndAction(
```

Comment:
I think this is where we will get a generic "Forbidden" error and we could make it more precise.
One thought: there could be more than one resource that the user cannot delete. You could collect a list of the names of the failures and pass that in the "details" of an exception. The details data is returned through the API in the `ErrorReport` object as a list of strings.